src/thread.ts

```diff
@@ -401,33 +382,47 @@ export class Thread<SCG extends ExtendableGenerics = DefaultGenerics> {
     }
   };
 
-  public markAsRead = () => {
-    const { read } = this.state.getLatestValue();
-    const currentUserId = this.client.user?.id;
-    const { unreadMessageCount } = (currentUserId && read[currentUserId]) || {};
-    if (!unreadMessageCount) return;
-    return this.channel.markRead({ thread_id: this.id });
+  public markAsRead = async () => {
+    const currentUserId = this.client.userID;
+
+    if (currentUserId && this.ownUnreadCount) {
+      const currentUserRead = this.state.getLatestValue().read[currentUserId];
+
+      if (currentUserRead) {
+        this.state.next((current) => ({
+          ...current,
+          read: {
+            ...current.read,
+            [currentUserId]: {
+              ...currentUserRead,
+              unreadMessageCount: 0,
+            },
+          },
+        }));
+      }
+
+      return await this.channel.markRead({ thread_id: this.id });
+    }
+
+    return null;
   };
 
-  // moved from channel to thread directly (skipped getClient thing as this call does not need active WS connection)
   public queryReplies = ({
-    sort = DEFAULT_SORT,
     limit = DEFAULT_PAGE_LIMIT,
+    sort = DEFAULT_SORT,
     ...otherOptions
   }: QueryRepliesOptions<SCG> = {}) => {
-    if (!this.channel) throw new Error('queryReplies: This Thread intance has no channel bound to it');
-
     return this.channel.getReplies(this.id, { limit, ...otherOptions }, sort);
   };
 
-  public loadPage = async (count: number) => {
+  public loadPage = async (count: number = DEFAULT_PAGE_LIMIT) => {
```

Comment:
Just tested this method and while I get why it exists I think it might be confusing to our integrators (pass -50 to count parameter to load previous fifty) - I suspect that this is a method that is going to be used within something like `loadPrevPage` and `loadNextPage` to deduplicate code? Also - in current state it always appends new messages no matter the direction.

Reply:
I'll keep loadNextPage/loadPrevPage, this method can be private instead.

I'll look into appending order!